src/services/helpers/getErc20Info.ts

```diff
@@ -0,0 +1,53 @@
+import Web3 from 'web3'
+
+import { Erc20Api, MinimalTokenDetails } from 'types'
+import { log } from 'utils'
+import { DEFAULT_PRECISION } from 'const'
+
+/**
+ * Wraps erc20 function and returns undefined in case of failure
+ */
+async function wrapPromise<T>(promise: Promise<T>): Promise<T | undefined> {
+  try {
+    return await promise
+  } catch (e) {
+    log('Failed to get ERC20 detail', e)
+    return
+  }
+}
+
+interface Params {
+  tokenAddress: string
+  erc20Api: Erc20Api
+  web3: Web3
+}
+
+/**
+ * Fetches info for an arbitrary ERC20 token from given address
+ */
+export async function getErc20Info({ tokenAddress, erc20Api, web3 }: Params): Promise<MinimalTokenDetails | null> {
```

Comment:
Would be good to cache all this on `tokenAddress, networkId`
Or will it be cached somewhere up the hierarchy?

Reply:
Don't completely get your question.

If you mean why this function does not follow the same pattern as the others, it's because this is a helper meant to be used internally by the services rather than directly in the ui/hooks.